.travis.yml

```diff
@@ -15,23 +15,30 @@ env:
     - secure: "cpOgenTtFAeKEdUBcynR1sZFsx7tSqKUytWAQ7oxDQgSbuHNhlazOfyQd9qytyKsc2jFZgfctGajAoZ/LvN71R21MxyQyFQDUg3RTsY9mNjg6GtcPyH+iUdIUxJkhEu/tdJqz5pzuz+hH8FGtTLP4A83N7En/apyAxQaJRL5A1Y="
     - secure: "WXquMki0USRcQ8vVvwKOk3mcUmoMufMLdDfbiIR/IZ9D1JL2Qu7/431BG9UBLDqs8OZXQRQNFrGhZpz6roczbcePZz7Wl5cdj8yd48WunkRpZDG1fIepHqppb3ydi/UfGXBPA6CaMAmjR1ErBchDl88ykQZE0L6mAdx4iyflOAg="
 
-before_install:
-  - travis_retry pip install codecov
-  - travis_retry npm install -g swagger-tools
-  - travis_retry curl -L -o cf.tgz "https://cli.run.pivotal.io/stable?release=linux64-binary&version=6.11.3&source=github-rel"
-  - tar xzvf cf.tgz
-  - export PATH=.:$PATH
-
```

Comment:
After reading up on some of the docs I'm curious to know why these moved from the `before_install` section to the other parts they are in now so I understand this better.  Is it fair to say that:

* It looks like most of these items are deploy related so their setup now happens right before actual deployment, which would cut down on setup time just to run the tests (especially if the tests fail).
* Since code coverage is just an assessment of the code itself and deploys are not contingent on them, this is now moved to after tests have succeeded but before deployment (are we doing anything else with code coverage other than getting a report of the metrics?).
* `swagger-tools` are just another dependency, so it moved to `before_script` with all of the other project dependencies.

Is this a correct assessment of the changes below?